GitHub_LearningTests/Feature/Login/Presentation/LoginBuilderTests.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  LoginBuilderTests.swift
+//  GitHub_LearningTests
+//
+//  Created by Amin Ashhab on 14.12.2020.
+//  Copyright © 2020 Amin_Second_Test_Project. All rights reserved.
+//
+
+import XCTest
+@testable import GitHub_Learning
+
+class LoginBuilderTests: XCTestCase {
+    private var sut: LoginBuilderImpl!
+    
+    func setUpTests() {
+        sut = LoginBuilderImpl()
+    }
+    
+    func test_givenEmailIsNotFilled_whenBuild_thenCredentailsAreNotBuild() {
+        // Given
+        setUpTests()
+        sut.email = nil
+        sut.password = "Test124567"
+        
+        // When
+        let credentails = try? sut.build()
+        
+        // Then
+        XCTAssertNil(credentails)
```

Comment:
Myslim ze je tam i assert na vyhozeny error. Pak by byla cast when a then spojena, ale bylo by to asi cistsi.